axiom/util_number_converter.c

```diff
@@ -46,6 +47,22 @@ void nr_itoa(char* buf, size_t len, int x) {
   nr_strlcpy(buf, p, len);
 }
 
+#define PRIORITY_MAXLEN 6
+char* nr_priority_double_to_str(double value) {
+  char* buf = NULL;
+
+  buf = nr_formatf("%.6f", value);
+
+  for (int i = 0; i < PRIORITY_MAXLEN; i++) {
+    if (',' == buf[i]) {
+      buf[i] = '.';
+      break;
+    }
+  }
+
+  return buf;
+}
+
```

Comment:
This function does not belong to this module: it is too distributed tracing specific. It knows specifics about how 'priority' should be formatted for tracestate header, not how priority (which btw is of type `nr_sampling_priority_t`) as double should be formatted in general. Maybe priority should be formatted differently in other contexts? This function should become a helper (`static`) function in `nr_distributed_trace.c` module and specifically be used to format priority for tracestate header.

Reply:
> Maybe priority should be formatted differently in other contexts?

I think that bespoke handling of priority formatting is part of the reason for this bug. If we're printing priority it should be standardized to a specific format, which was my rationale for making this a utility function- that way if there are other places in the code that handle the priority -> string conversion incorrectly, we can directly reference this function. It would also serve as a template for any priority handling code going forward that I think may be lost if hidden in `nr_distributed_trace.c`. Declaring it as a utility function also allows us to directly call it in any axiom unit tests (or debug log messages) if need be outside of that file.

I agree that it's a little too specific perhaps to live in this util file and maybe should be relocated, but would still like to avoid making it `static` and exist solely in `nr_distributed_trace.c`.

Reply:
My point is that priority in distributed tracing's New Relic's W3C tracestate needs a '.' as radix separator and 6 decimal digits - this is very specific functionality. Maybe priority needs to be formatted differently in other tracing header - I don't know all the use cases and formats. This use case is known and fairly well defined and is local to distributed tracing's New Relic's W3C tracestate.

Reply:
I'll do a quick sweep of the code and see if this function is useful anywhere else and implement your suggestion if not

Reply:
I think most priority fields in JSON are created via `nro_to_json()` for other usages in the code.  This used `%.5f` fwiw.

Reply:
I think this would be a good helper function in `util_sampling.c` which acts as a conversion function from the native data type to the accepted string representation.

Reply:
function declaration moved in this commit: 880b18e98ed259ea876d70bc91e43a018c0b949f
`debug_dt` flag added in this commit: 29a2c65d0e577e32ab2c14300173255e2e6773bc
format string addressed in this commit: f37e684cc8925737b8a5664750f179975032d937